### PR TITLE
feat(worker): setup rabbitmq and worker

### PR DESCRIPTION
**Pull request by sweeetland**
# Context

- ADP1149

# Proposed Solution

Setup RabbitMQ in chain follower package. Create a new package called worker. The worker package should enable us to test the chain-follower and worker in the same process, via `InMemoryClient`. We also have a `RabbitMqClient` that relies on RabbitMQ.

### CODE REVIEW

File: packages/chain-follower/docker-compose.yml [hunk 1]
```diff
@@ -1,6 +1,15 @@
 version: "3.5"
 
 services:
+  rabbitmq:
+    image: rabbitmq:3-management
+    container_name: 'rabbitmq'
```
Comment (rhyslbw) [inline packages/chain-follower/docker-compose.yml:6]: This is superfluous, as the service name is used by default.
```suggestion

```

File: packages/worker/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { cleanEnv, EnvError, makeValidator, Spec } from 'envalid'
+
+// custom validator same as str from envalid but with typed return type
+const str = <T extends string = string> (spec?: Spec<T>) => makeValidator((input: string) => {
+  if (typeof input === 'string') return input as T
+  throw new EnvError(`Not a string: "${input}"`)
+})(spec)
+
+type LogLevelString = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'fatal';
```
Comment (rhyslbw) [inline packages/worker/src/config.ts:9]: Is this not available from the `ts-log` package?

File: packages/worker/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+(async () => {
+  const config = getConfig()
+
+  const logger: Logger = createLogger({
+    name: 'worker',
+    level: config.loggerMinSeverity
+  })
+
+  try {
+    const queue = await RabbitMqClient()
```
Comment (rhyslbw) [inline packages/worker/src/index.ts:17]:
```suggestion
    const queueClient = await RabbitMqClient()
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/worker/src/config.ts (hunk 2 above)
Comment (sweeetland) [reply packages/worker/src/config.ts]: ah yes it's available from `bunyan` package which is what I'm guessing you meant.

Comment (rhyslbw) [reply packages/worker/src/config.ts]: I did expect it to be in the ts-log package, but bunyan is fine to source from too.

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**